// @azure/ai-projects/dist/commonjs/classic/datasets/index.js
"use strict";
// Copyright (c) Microsoft Corporation.
// Licensed under the MIT License.
/* eslint-disable tsdoc/syntax */
Object.defineProperty(exports, "__esModule", { value: true });
exports._getDatasetsOperations = _getDatasetsOperations;
const operations_js_1 = require("../../api/datasets/operations.js");
function _getDatasets(context) {
    return {
        getCredentials: (name, version, options) => (0, operations_js_1.getCredentials)(context, name, version, options),
        pendingUpload: (name, version, body, options) => (0, operations_js_1.pendingUpload)(context, name, version, body, options),
        createOrUpdate: (name, version, body, options) => (0, operations_js_1.createOrUpdate)(context, name, version, body, options),
        delete: (name, version, options) => (0, operations_js_1.$delete)(context, name, version, options),
        get: (name, version, options) => (0, operations_js_1.get)(context, name, version, options),
        list: (options) => (0, operations_js_1.list)(context, options),
        listVersions: (name, options) => (0, operations_js_1.listVersions)(context, name, options),
        uploadFile: (name, version, filePath, connectionName) => (0, operations_js_1.uploadFile)(context, name, version, filePath, connectionName),
        uploadFolder: (name, version, folderPath, connectionName) => (0, operations_js_1.uploadFolder)(context, name, version, folderPath, connectionName),
    };
}
function _getDatasetsOperations(context) {
    return Object.assign({}, _getDatasets(context));
}
