// @azure/ai-projects/dist/react-native/api/evaluations/operations.js
// Copyright (c) Microsoft Corporation.
// Licensed under the MIT License.
import { evaluationSerializer, evaluationDeserializer, _pagedEvaluationDeserializer, agentEvaluationRequestSerializer, agentEvaluationDeserializer, } from "../../models/models.js";
import { expandUrlTemplate } from "../../static-helpers/urlTemplate.js";
import { buildPagedAsyncIterator, } from "../../static-helpers/pagingHelpers.js";
import { createRestError, operationOptionsToRequestParameters, } from "@azure-rest/core-client";
export function _createAgentEvaluationSend(context, evaluation, options = {
    requestOptions: {},
}) {
    var _a, _b;
    const path = expandUrlTemplate("/evaluations/runs:runAgent{?api%2Dversion}", {
        "api%2Dversion": context.apiVersion,
    }, {
        allowReserved: (_a = options === null || options === void 0 ? void 0 : options.requestOptions) === null || _a === void 0 ? void 0 : _a.skipUrlEncoding,
    });
    return context.path(path).post(Object.assign(Object.assign({}, operationOptionsToRequestParameters(options)), { contentType: "application/json", headers: Object.assign({ accept: "application/json" }, (_b = options.requestOptions) === null || _b === void 0 ? void 0 : _b.headers), body: agentEvaluationRequestSerializer(evaluation) }));
}
export async function _createAgentEvaluationDeserialize(result) {
    const expectedStatuses = ["201"];
    if (!expectedStatuses.includes(result.status)) {
        throw createRestError(result);
    }
    return agentEvaluationDeserializer(result.body);
}
/** Creates an agent evaluation run. */
export async function createAgentEvaluation(context, evaluation, options = {
    requestOptions: {},
}) {
    const result = await _createAgentEvaluationSend(context, evaluation, options);
    return _createAgentEvaluationDeserialize(result);
}
export function _createSend(context, evaluation, options = { requestOptions: {} }) {
    var _a, _b;
    const path = expandUrlTemplate("/evaluations/runs:run{?api-version}", {
        "api-version": context.apiVersion,
    }, {
        allowReserved: (_a = options === null || options === void 0 ? void 0 : options.requestOptions) === null || _a === void 0 ? void 0 : _a.skipUrlEncoding,
    });
    return context.path(path).post(Object.assign(Object.assign({}, operationOptionsToRequestParameters(options)), { contentType: "application/json", headers: Object.assign({ accept: "application/json" }, (_b = options.requestOptions) === null || _b === void 0 ? void 0 : _b.headers), body: evaluationSerializer(evaluation) }));
}
export async function _createDeserialize(result) {
    const expectedStatuses = ["201"];
    if (!expectedStatuses.includes(result.status)) {
        throw createRestError(result);
    }
    return evaluationDeserializer(result.body);
}
/** Creates an evaluation run. */
export async function create(context, evaluation, options = { requestOptions: {} }) {
    const result = await _createSend(context, evaluation, options);
    return _createDeserialize(result);
}
export function _listSend(context, options = { requestOptions: {} }) {
    var _a, _b;
    const path = expandUrlTemplate("/evaluations/runs{?api-version}", {
        "api-version": context.apiVersion,
    }, {
        allowReserved: (_a = options === null || options === void 0 ? void 0 : options.requestOptions) === null || _a === void 0 ? void 0 : _a.skipUrlEncoding,
    });
    return context.path(path).get(Object.assign(Object.assign({}, operationOptionsToRequestParameters(options)), { headers: Object.assign(Object.assign(Object.assign({}, ((options === null || options === void 0 ? void 0 : options.clientRequestId) !== undefined
            ? { "x-ms-client-request-id": options === null || options === void 0 ? void 0 : options.clientRequestId }
            : {})), { accept: "application/json" }), (_b = options.requestOptions) === null || _b === void 0 ? void 0 : _b.headers) }));
}
export async function _listDeserialize(result) {
    const expectedStatuses = ["200"];
    if (!expectedStatuses.includes(result.status)) {
        throw createRestError(result);
    }
    return _pagedEvaluationDeserializer(result.body);
}
/** List evaluation runs */
export function list(context, options = { requestOptions: {} }) {
    return buildPagedAsyncIterator(context, () => _listSend(context, options), _listDeserialize, ["200"], { itemName: "value", nextLinkName: "nextLink" });
}
export function _getSend(context, name, options = { requestOptions: {} }) {
    var _a, _b;
    const path = expandUrlTemplate("/evaluations/runs/{name}{?api-version}", {
        name: name,
        "api-version": context.apiVersion,
    }, {
        allowReserved: (_a = options === null || options === void 0 ? void 0 : options.requestOptions) === null || _a === void 0 ? void 0 : _a.skipUrlEncoding,
    });
    return context.path(path).get(Object.assign(Object.assign({}, operationOptionsToRequestParameters(options)), { headers: Object.assign(Object.assign(Object.assign({}, ((options === null || options === void 0 ? void 0 : options.clientRequestId) !== undefined
            ? { "x-ms-client-request-id": options === null || options === void 0 ? void 0 : options.clientRequestId }
            : {})), { accept: "application/json" }), (_b = options.requestOptions) === null || _b === void 0 ? void 0 : _b.headers) }));
}
export async function _getDeserialize(result) {
    const expectedStatuses = ["200"];
    if (!expectedStatuses.includes(result.status)) {
        throw createRestError(result);
    }
    return evaluationDeserializer(result.body);
}
/** Get an evaluation run by name. */
export async function get(context, name, options = { requestOptions: {} }) {
    const result = await _getSend(context, name, options);
    return _getDeserialize(result);
}
