// @azure/ai-projects/dist/react-native/models/models.js
// Copyright (c) Microsoft Corporation.
// Licensed under the MIT License.
/* eslint-disable @typescript-eslint/explicit-module-boundary-types */
/* eslint-disable @typescript-eslint/naming-convention */
export function connectionDeserializer(item) {
    return {
        name: item["name"],
        id: item["id"],
        type: item["type"],
        target: item["target"],
        metadata: item["metadata"],
        isDefault: item["isDefault"],
        credentials: item["credentials"]
            ? baseCredentialsUnionDeserializer(item["credentials"])
            : { type: "None" },
    };
}
export function baseCredentialsDeserializer(item) {
    return {
        type: item["type"],
    };
}
export function baseCredentialsUnionDeserializer(item) {
    switch (item.type) {
        case "ApiKey":
            return apiKeyCredentialsDeserializer(item);
        case "AAD":
            return entraIDCredentialsDeserializer(item);
        case "CustomKeys":
            return customCredentialDeserializer(item);
        case "SAS":
            return sasCredentialsDeserializer(item);
        case "None":
            return noAuthenticationCredentialsDeserializer(item);
        default:
            return baseCredentialsDeserializer(item);
    }
}
export function apiKeyCredentialsDeserializer(item) {
    return {
        type: item["type"],
        apiKey: item["key"],
    };
}
export function entraIDCredentialsDeserializer(item) {
    return {
        type: item["type"],
    };
}
export function customCredentialDeserializer(item) {
    return {
        type: item["type"],
        keys: item["keys"],
    };
}
export function sasCredentialsDeserializer(item) {
    return {
        type: item["type"],
        sasToken: item["SAS"],
    };
}
export function noAuthenticationCredentialsDeserializer(item) {
    return {
        type: item["type"],
    };
}
export function _pagedConnectionDeserializer(item) {
    return {
        value: connectionArrayDeserializer(item["value"]),
        nextLink: item["nextLink"],
    };
}
export function connectionArrayDeserializer(result) {
    return result.map((item) => {
        return connectionDeserializer(item);
    });
}
// optional name property
export function evaluationSerializer(item) {
    return {
        data: inputDataUnionSerializer(item["data"]),
        displayName: item["displayName"],
        description: item["description"],
        tags: item["tags"],
        properties: item["properties"],
        evaluators: evaluatorConfigurationRecordSerializer(item["evaluators"]),
    };
}
export function evaluationDeserializer(item) {
    return {
        name: item["id"],
        data: inputDataUnionDeserializer(item["data"]),
        displayName: item["displayName"],
        description: item["description"],
        status: item["status"],
        tags: item["tags"],
        properties: item["properties"],
        evaluators: evaluatorConfigurationRecordDeserializer(item["evaluators"]),
    };
}
export function inputDataSerializer(item) {
    return { type: item["type"] };
}
export function inputDataDeserializer(item) {
    return {
        type: item["type"],
    };
}
export function inputDataUnionSerializer(item) {
    switch (item.type) {
        case "dataset":
            return inputDatasetSerializer(item);
        default:
            return inputDataSerializer(item);
    }
}
export function inputDataUnionDeserializer(_item) {
    const item = _item || {};
    switch (item.type) {
        case "dataset":
            return inputDatasetDeserializer(item);
        default:
            return inputDataDeserializer(item);
    }
}
export function inputDatasetSerializer(item) {
    return { type: item["type"], id: item["id"] };
}
export function inputDatasetDeserializer(item) {
    return {
        type: item["type"],
        id: item["id"],
    };
}
export function evaluatorConfigurationRecordSerializer(item) {
    const result = {};
    Object.keys(item).map((key) => {
        result[key] = !item[key] ? item[key] : evaluatorConfigurationSerializer(item[key]);
    });
    return result;
}
export function evaluatorConfigurationRecordDeserializer(item) {
    const result = {};
    Object.keys(item).map((key) => {
        result[key] = !item[key] ? item[key] : evaluatorConfigurationDeserializer(item[key]);
    });
    return result;
}
export function evaluatorConfigurationSerializer(item) {
    return {
        id: item["id"],
        initParams: item["initParams"],
        dataMapping: item["dataMapping"],
    };
}
export function evaluatorConfigurationDeserializer(item) {
    return {
        id: item["id"],
        initParams: item["initParams"],
        dataMapping: item["dataMapping"],
    };
}
export function _pagedEvaluationDeserializer(item) {
    return {
        value: evaluationArrayDeserializer(item["value"]),
        nextLink: item["nextLink"],
    };
}
export function evaluationArraySerializer(result) {
    return result.map((item) => {
        return evaluationSerializer(item);
    });
}
export function evaluationArrayDeserializer(result) {
    return result.map((item) => {
        return evaluationDeserializer(item);
    });
}
export function agentEvaluationRequestSerializer(item) {
    return {
        runId: item["runId"],
        threadId: item["threadId"],
        evaluators: evaluatorConfigurationRecordSerializer(item["evaluators"]),
        samplingConfiguration: !item["samplingConfiguration"]
            ? item["samplingConfiguration"]
            : agentEvaluationSamplingConfigurationSerializer(item["samplingConfiguration"]),
        redactionConfiguration: !item["redactionConfiguration"]
            ? item["redactionConfiguration"]
            : agentEvaluationRedactionConfigurationSerializer(item["redactionConfiguration"]),
        appInsightsConnectionString: item["appInsightsConnectionString"],
    };
}
export function agentEvaluationSamplingConfigurationSerializer(item) {
    return {
        name: item["name"],
        samplingPercent: item["samplingPercent"],
        maxRequestRate: item["maxRequestRate"],
    };
}
export function agentEvaluationRedactionConfigurationSerializer(item) {
    return { redactScoreProperties: item["redactScoreProperties"] };
}
export function agentEvaluationDeserializer(item) {
    return {
        id: item["id"],
        status: item["status"],
        error: item["error"],
        result: !item["result"]
            ? item["result"]
            : agentEvaluationResultArrayDeserializer(item["result"]),
    };
}
export function agentEvaluationResultArrayDeserializer(result) {
    return result.map((item) => {
        return agentEvaluationResultDeserializer(item);
    });
}
export function agentEvaluationResultDeserializer(item) {
    return {
        evaluator: item["evaluator"],
        evaluatorId: item["evaluatorId"],
        score: item["score"],
        status: item["status"],
        reason: item["reason"],
        version: item["version"],
        threadId: item["threadId"],
        runId: item["runId"],
        error: item["error"],
        additionalDetails: item["additionalDetails"],
    };
}
export function _pagedDatasetVersionDeserializer(item) {
    return {
        value: datasetVersionUnionArrayDeserializer(item["value"]),
        nextLink: item["nextLink"],
    };
}
export function datasetVersionUnionArraySerializer(result) {
    return result.map((item) => {
        return datasetVersionUnionSerializer(item);
    });
}
export function datasetVersionUnionArrayDeserializer(result) {
    return result.map((item) => {
        return datasetVersionUnionDeserializer(item);
    });
}
export function datasetVersionSerializer(item) {
    return {
        dataUri: item["dataUri"],
        type: item["type"],
        connectionName: item["connectionName"],
        description: item["description"],
        tags: item["tags"],
    };
}
export function datasetVersionDeserializer(item) {
    return {
        dataUri: item["dataUri"],
        type: item["type"],
        isReference: item["isReference"],
        connectionName: item["connectionName"],
        id: item["id"],
        name: item["name"],
        version: item["version"],
        description: item["description"],
        tags: item["tags"],
    };
}
export function datasetVersionUnionSerializer(item) {
    switch (item.type) {
        case "uri_file":
            return fileDatasetVersionSerializer(item);
        case "uri_folder":
            return folderDatasetVersionSerializer(item);
        default:
            return datasetVersionSerializer(item);
    }
}
export function datasetVersionUnionDeserializer(item) {
    switch (item.type) {
        case "uri_file":
            return fileDatasetVersionDeserializer(item);
        case "uri_folder":
            return folderDatasetVersionDeserializer(item);
        default:
            return datasetVersionDeserializer(item);
    }
}
export function fileDatasetVersionSerializer(item) {
    return {
        dataUri: item["dataUri"],
        type: item["type"],
        connectionName: item["connectionName"],
        description: item["description"],
        tags: item["tags"],
    };
}
export function fileDatasetVersionDeserializer(item) {
    return {
        dataUri: item["dataUri"],
        type: item["type"],
        isReference: item["isReference"],
        connectionName: item["connectionName"],
        id: item["id"],
        name: item["name"],
        version: item["version"],
        description: item["description"],
        tags: item["tags"],
    };
}
export function folderDatasetVersionSerializer(item) {
    return {
        dataUri: item["dataUri"],
        type: item["type"],
        connectionName: item["connectionName"],
        description: item["description"],
        tags: item["tags"],
    };
}
export function folderDatasetVersionDeserializer(item) {
    return {
        dataUri: item["dataUri"],
        type: item["type"],
        isReference: item["isReference"],
        connectionName: item["connectionName"],
        id: item["id"],
        name: item["name"],
        version: item["version"],
        description: item["description"],
        tags: item["tags"],
    };
}
export function pendingUploadRequestSerializer(item) {
    return {
        pendingUploadId: item["pendingUploadId"],
        connectionName: item["connectionName"],
        pendingUploadType: item["pendingUploadType"],
    };
}
export function pendingUploadResponseDeserializer(item) {
    return {
        blobReference: blobReferenceDeserializer(item["blobReference"]),
        pendingUploadId: item["pendingUploadId"],
        version: item["version"],
        pendingUploadType: item["pendingUploadType"],
    };
}
export function blobReferenceDeserializer(item) {
    return {
        blobUri: item["blobUri"],
        storageAccountArmId: item["storageAccountArmId"],
        credential: sasCredentialDeserializer(item["credential"]),
    };
}
export function sasCredentialDeserializer(item) {
    return {
        sasUri: item["sasUri"],
        type: item["type"],
    };
}
export function assetCredentialResponseDeserializer(item) {
    return {
        blobReference: blobReferenceDeserializer(item["blobReference"]),
    };
}
export function _pagedIndexDeserializer(item) {
    return {
        value: indexUnionArrayDeserializer(item["value"]),
        nextLink: item["nextLink"],
    };
}
export function indexUnionArraySerializer(result) {
    return result.map((item) => {
        return indexUnionSerializer(item);
    });
}
export function indexUnionArrayDeserializer(result) {
    return result.map((item) => {
        return indexUnionDeserializer(item);
    });
}
export function indexSerializer(item) {
    return {
        type: item["type"],
        description: item["description"],
        tags: item["tags"],
    };
}
export function indexDeserializer(item) {
    return {
        type: item["type"],
        id: item["id"],
        name: item["name"],
        version: item["version"],
        description: item["description"],
        tags: item["tags"],
    };
}
export function indexUnionSerializer(item) {
    switch (item.type) {
        case "AzureSearch":
            return azureAISearchIndexSerializer(item);
        case "ManagedAzureSearch":
            return managedAzureAISearchIndexSerializer(item);
        case "CosmosDBNoSqlVectorStore":
            return cosmosDBIndexSerializer(item);
        default:
            return indexSerializer(item);
    }
}
export function indexUnionDeserializer(item) {
    switch (item.type) {
        case "AzureSearch":
            return azureAISearchIndexDeserializer(item);
        case "ManagedAzureSearch":
            return managedAzureAISearchIndexDeserializer(item);
        case "CosmosDBNoSqlVectorStore":
            return cosmosDBIndexDeserializer(item);
        default:
            return indexDeserializer(item);
    }
}
export function azureAISearchIndexSerializer(item) {
    return {
        type: item["type"],
        description: item["description"],
        tags: item["tags"],
        connectionName: item["connectionName"],
        indexName: item["indexName"],
        fieldMapping: !item["fieldMapping"]
            ? item["fieldMapping"]
            : fieldMappingSerializer(item["fieldMapping"]),
    };
}
export function azureAISearchIndexDeserializer(item) {
    return {
        type: item["type"],
        id: item["id"],
        name: item["name"],
        version: item["version"],
        description: item["description"],
        tags: item["tags"],
        connectionName: item["connectionName"],
        indexName: item["indexName"],
        fieldMapping: !item["fieldMapping"]
            ? item["fieldMapping"]
            : fieldMappingDeserializer(item["fieldMapping"]),
    };
}
export function fieldMappingSerializer(item) {
    return {
        contentFields: item["contentFields"].map((p) => {
            return p;
        }),
        filepathField: item["filepathField"],
        titleField: item["titleField"],
        urlField: item["urlField"],
        vectorFields: !item["vectorFields"]
            ? item["vectorFields"]
            : item["vectorFields"].map((p) => {
                return p;
            }),
        metadataFields: !item["metadataFields"]
            ? item["metadataFields"]
            : item["metadataFields"].map((p) => {
                return p;
            }),
    };
}
export function fieldMappingDeserializer(item) {
    return {
        contentFields: item["contentFields"].map((p) => {
            return p;
        }),
        filepathField: item["filepathField"],
        titleField: item["titleField"],
        urlField: item["urlField"],
        vectorFields: !item["vectorFields"]
            ? item["vectorFields"]
            : item["vectorFields"].map((p) => {
                return p;
            }),
        metadataFields: !item["metadataFields"]
            ? item["metadataFields"]
            : item["metadataFields"].map((p) => {
                return p;
            }),
    };
}
export function managedAzureAISearchIndexSerializer(item) {
    return {
        type: item["type"],
        description: item["description"],
        tags: item["tags"],
        vectorStoreId: item["vectorStoreId"],
    };
}
export function managedAzureAISearchIndexDeserializer(item) {
    return {
        type: item["type"],
        id: item["id"],
        name: item["name"],
        version: item["version"],
        description: item["description"],
        tags: item["tags"],
        vectorStoreId: item["vectorStoreId"],
    };
}
export function cosmosDBIndexSerializer(item) {
    return {
        type: item["type"],
        description: item["description"],
        tags: item["tags"],
        connectionName: item["connectionName"],
        databaseName: item["databaseName"],
        containerName: item["containerName"],
        embeddingConfiguration: embeddingConfigurationSerializer(item["embeddingConfiguration"]),
        fieldMapping: fieldMappingSerializer(item["fieldMapping"]),
    };
}
export function cosmosDBIndexDeserializer(item) {
    return {
        type: item["type"],
        id: item["id"],
        name: item["name"],
        version: item["version"],
        description: item["description"],
        tags: item["tags"],
        connectionName: item["connectionName"],
        databaseName: item["databaseName"],
        containerName: item["containerName"],
        embeddingConfiguration: embeddingConfigurationDeserializer(item["embeddingConfiguration"]),
        fieldMapping: fieldMappingDeserializer(item["fieldMapping"]),
    };
}
export function embeddingConfigurationSerializer(item) {
    return {
        modelDeploymentName: item["modelDeploymentName"],
        embeddingField: item["embeddingField"],
    };
}
export function embeddingConfigurationDeserializer(item) {
    return {
        modelDeploymentName: item["modelDeploymentName"],
        embeddingField: item["embeddingField"],
    };
}
export function deploymentDeserializer(item) {
    return {
        type: item["type"],
        name: item["name"],
    };
}
export function deploymentUnionDeserializer(item) {
    switch (item.type) {
        case "ModelDeployment":
            return modelDeploymentDeserializer(item);
        default:
            return deploymentDeserializer(item);
    }
}
export function modelDeploymentDeserializer(item) {
    return {
        type: item["type"],
        name: item["name"],
        modelName: item["modelName"],
        modelVersion: item["modelVersion"],
        modelPublisher: item["modelPublisher"],
        capabilities: item["capabilities"],
        sku: skuDeserializer(item["sku"]),
        connectionName: item["connectionName"],
    };
}
export function skuDeserializer(item) {
    return {
        capacity: item["capacity"],
        family: item["family"],
        name: item["name"],
        size: item["size"],
        tier: item["tier"],
    };
}
export function _pagedDeploymentDeserializer(item) {
    return {
        value: deploymentUnionArrayDeserializer(item["value"]),
        nextLink: item["nextLink"],
    };
}
export function deploymentUnionArrayDeserializer(result) {
    return result.map((item) => {
        return deploymentUnionDeserializer(item);
    });
}
export function redTeamSerializer(item) {
    return {
        displayName: item["displayName"],
        numTurns: item["numTurns"],
        attackStrategies: !item["attackStrategies"]
            ? item["attackStrategies"]
            : item["attackStrategies"].map((p) => {
                return p;
            }),
        simulationOnly: item["simulationOnly"],
        riskCategories: !item["riskCategories"]
            ? item["riskCategories"]
            : item["riskCategories"].map((p) => {
                return p;
            }),
        applicationScenario: item["applicationScenario"],
        tags: item["tags"],
        properties: item["properties"],
        target: targetConfigUnionSerializer(item["target"]),
    };
}
export function redTeamDeserializer(item) {
    return {
        name: item["id"],
        displayName: item["displayName"],
        numTurns: item["numTurns"],
        attackStrategies: !item["attackStrategies"]
            ? item["attackStrategies"]
            : item["attackStrategies"].map((p) => {
                return p;
            }),
        simulationOnly: item["simulationOnly"],
        riskCategories: !item["riskCategories"]
            ? item["riskCategories"]
            : item["riskCategories"].map((p) => {
                return p;
            }),
        applicationScenario: item["applicationScenario"],
        tags: item["tags"],
        properties: item["properties"],
        status: item["status"],
        target: targetConfigUnionDeserializer(item["target"]),
    };
}
export function targetConfigSerializer(item) {
    return { type: item["type"] };
}
export function targetConfigDeserializer(item) {
    return {
        type: item["type"],
    };
}
export function targetConfigUnionSerializer(item) {
    switch (item.type) {
        case "AzureOpenAIModel":
            return azureOpenAIModelConfigurationSerializer(item);
        default:
            return targetConfigSerializer(item);
    }
}
export function targetConfigUnionDeserializer(item) {
    switch (item.type) {
        case "AzureOpenAIModel":
            return azureOpenAIModelConfigurationDeserializer(item);
        default:
            return targetConfigDeserializer(item);
    }
}
export function azureOpenAIModelConfigurationSerializer(item) {
    return { modelDeploymentName: item["modelDeploymentName"] };
}
export function azureOpenAIModelConfigurationDeserializer(item) {
    return {
        type: item["type"],
        modelDeploymentName: item["modelDeploymentName"],
    };
}
export function _pagedRedTeamDeserializer(item) {
    return {
        value: redTeamArrayDeserializer(item["value"]),
        nextLink: item["nextLink"],
    };
}
export function redTeamArraySerializer(result) {
    return result.map((item) => {
        return redTeamSerializer(item);
    });
}
export function redTeamArrayDeserializer(result) {
    return result.map((item) => {
        return redTeamDeserializer(item);
    });
}
/** Azure AI Projects API versions */
export var KnownVersions;
(function (KnownVersions) {
    /** Azure AI API version 2025-05-01. */
    KnownVersions["V20250501"] = "2025-05-01";
    /** Azure AI API version 2025-05-15-preview. */
    KnownVersions["V20250515Preview"] = "2025-05-15-preview";
})(KnownVersions || (KnownVersions = {}));
