// @azure/ai-agents/dist/esm/api/operations.js
// Copyright (c) Microsoft Corporation.
// Licensed under the MIT License.
import { __asyncGenerator, __asyncValues, __await } from "tslib";
import { toolResourcesSerializer, toolDefinitionUnionArraySerializer, agentsResponseFormatOptionSerializer, agentDeserializer, _agentsPagedResultAgentDeserializer, agentDeletionStatusDeserializer, agentThreadCreationOptionsSerializer, updateToolResourcesOptionsSerializer, truncationObjectSerializer, agentsToolChoiceOptionSerializer, threadRunDeserializer, } from "../models/models.js";
import { buildPagedAsyncIterator, } from "../static-helpers/pagingHelpers.js";
import { expandUrlTemplate } from "../static-helpers/urlTemplate.js";
import { createRestError, operationOptionsToRequestParameters, } from "@azure-rest/core-client";
import { MessageStreamEvent, RunStepStreamEvent, RunStreamEvent, ThreadStreamEvent, } from "../models/streamingModels.js";
import { createSseStream } from "@azure/core-sse";
import { isNodeLike } from "@azure/core-util";
import { logger } from "../logger.js";
import { _createRunSend } from "./runs/operations.js";
export function _createThreadAndRunSend(context, assistantId, options = { requestOptions: {} }) {
    var _a, _b;
    const path = expandUrlTemplate("/threads/runs{?api%2Dversion}", {
        "api%2Dversion": context.apiVersion,
    }, {
        allowReserved: (_a = options === null || options === void 0 ? void 0 : options.requestOptions) === null || _a === void 0 ? void 0 : _a.skipUrlEncoding,
    });
    return context.path(path).post(Object.assign(Object.assign({}, operationOptionsToRequestParameters(options)), { contentType: "application/json", headers: Object.assign({ accept: "application/json" }, (_b = options.requestOptions) === null || _b === void 0 ? void 0 : _b.headers), body: {
            assistant_id: assistantId,
            thread: !(options === null || options === void 0 ? void 0 : options.thread)
                ? options === null || options === void 0 ? void 0 : options.thread
                : agentThreadCreationOptionsSerializer(options === null || options === void 0 ? void 0 : options.thread),
            model: options === null || options === void 0 ? void 0 : options.model,
            instructions: options === null || options === void 0 ? void 0 : options.instructions,
            tools: !(options === null || options === void 0 ? void 0 : options.tools) ? options === null || options === void 0 ? void 0 : options.tools : toolDefinitionUnionArraySerializer(options === null || options === void 0 ? void 0 : options.tools),
            tool_resources: !(options === null || options === void 0 ? void 0 : options.toolResources)
                ? options === null || options === void 0 ? void 0 : options.toolResources
                : updateToolResourcesOptionsSerializer(options === null || options === void 0 ? void 0 : options.toolResources),
            stream: options === null || options === void 0 ? void 0 : options.stream,
            temperature: options === null || options === void 0 ? void 0 : options.temperature,
            top_p: options === null || options === void 0 ? void 0 : options.topP,
            max_prompt_tokens: options === null || options === void 0 ? void 0 : options.maxPromptTokens,
            max_completion_tokens: options === null || options === void 0 ? void 0 : options.maxCompletionTokens,
            truncation_strategy: !(options === null || options === void 0 ? void 0 : options.truncationStrategy)
                ? options === null || options === void 0 ? void 0 : options.truncationStrategy
                : truncationObjectSerializer(options === null || options === void 0 ? void 0 : options.truncationStrategy),
            tool_choice: !(options === null || options === void 0 ? void 0 : options.toolChoice)
                ? options === null || options === void 0 ? void 0 : options.toolChoice
                : agentsToolChoiceOptionSerializer(options === null || options === void 0 ? void 0 : options.toolChoice),
            response_format: !(options === null || options === void 0 ? void 0 : options.responseFormat)
                ? options === null || options === void 0 ? void 0 : options.responseFormat
                : agentsResponseFormatOptionSerializer(options === null || options === void 0 ? void 0 : options.responseFormat),
            parallel_tool_calls: options === null || options === void 0 ? void 0 : options.parallelToolCalls,
            metadata: options === null || options === void 0 ? void 0 : options.metadata,
        } }));
}
export async function _createThreadAndRunDeserialize(result) {
    const expectedStatuses = ["200"];
    if (!expectedStatuses.includes(result.status)) {
        throw createRestError(result);
    }
    return threadRunDeserializer(result.body);
}
/** Creates a new agent thread and immediately starts a run using that new thread. */
export function createThreadAndRun(context, assistantId, options = { requestOptions: {} }) {
    async function executeCreateThreadAndRun() {
        const result = await _createThreadAndRunSend(context, assistantId, options);
        return _createThreadAndRunDeserialize(result);
    }
    return {
        then: function (onFulfilled, onRejected) {
            return executeCreateThreadAndRun().then(onFulfilled, onRejected).catch(onRejected);
        },
        async stream() {
            return createThreadAndRunStreaming(context, assistantId, options);
        },
    };
}
export function _deleteAgentSend(context, assistantId, options = { requestOptions: {} }) {
    var _a, _b;
    const path = expandUrlTemplate("/assistants/{assistantId}{?api%2Dversion}", {
        assistantId: assistantId,
        "api%2Dversion": context.apiVersion,
    }, {
        allowReserved: (_a = options === null || options === void 0 ? void 0 : options.requestOptions) === null || _a === void 0 ? void 0 : _a.skipUrlEncoding,
    });
    return context.path(path).delete(Object.assign(Object.assign({}, operationOptionsToRequestParameters(options)), { headers: Object.assign({ accept: "application/json" }, (_b = options.requestOptions) === null || _b === void 0 ? void 0 : _b.headers) }));
}
export async function _deleteAgentDeserialize(result) {
    const expectedStatuses = ["200"];
    if (!expectedStatuses.includes(result.status)) {
        throw createRestError(result);
    }
    return agentDeletionStatusDeserializer(result.body);
}
/** Deletes an agent. */
export async function deleteAgent(context, assistantId, options = { requestOptions: {} }) {
    const result = await _deleteAgentSend(context, assistantId, options);
    return _deleteAgentDeserialize(result);
}
export function _updateAgentSend(context, assistantId, options = { requestOptions: {} }) {
    var _a, _b;
    const path = expandUrlTemplate("/assistants/{assistantId}{?api%2Dversion}", {
        assistantId: assistantId,
        "api%2Dversion": context.apiVersion,
    }, {
        allowReserved: (_a = options === null || options === void 0 ? void 0 : options.requestOptions) === null || _a === void 0 ? void 0 : _a.skipUrlEncoding,
    });
    return context.path(path).post(Object.assign(Object.assign({}, operationOptionsToRequestParameters(options)), { contentType: "application/json", headers: Object.assign({ accept: "application/json" }, (_b = options.requestOptions) === null || _b === void 0 ? void 0 : _b.headers), body: {
            model: options === null || options === void 0 ? void 0 : options.model,
            name: options === null || options === void 0 ? void 0 : options.name,
            description: options === null || options === void 0 ? void 0 : options.description,
            instructions: options === null || options === void 0 ? void 0 : options.instructions,
            tools: !(options === null || options === void 0 ? void 0 : options.tools) ? options === null || options === void 0 ? void 0 : options.tools : toolDefinitionUnionArraySerializer(options === null || options === void 0 ? void 0 : options.tools),
            tool_resources: !(options === null || options === void 0 ? void 0 : options.toolResources)
                ? options === null || options === void 0 ? void 0 : options.toolResources
                : toolResourcesSerializer(options === null || options === void 0 ? void 0 : options.toolResources),
            temperature: options === null || options === void 0 ? void 0 : options.temperature,
            top_p: options === null || options === void 0 ? void 0 : options.topP,
            response_format: !(options === null || options === void 0 ? void 0 : options.responseFormat)
                ? options === null || options === void 0 ? void 0 : options.responseFormat
                : agentsResponseFormatOptionSerializer(options === null || options === void 0 ? void 0 : options.responseFormat),
            metadata: options === null || options === void 0 ? void 0 : options.metadata,
        } }));
}
export async function _updateAgentDeserialize(result) {
    const expectedStatuses = ["200"];
    if (!expectedStatuses.includes(result.status)) {
        throw createRestError(result);
    }
    return agentDeserializer(result.body);
}
/** Modifies an existing agent. */
export async function updateAgent(context, assistantId, options = { requestOptions: {} }) {
    const result = await _updateAgentSend(context, assistantId, options);
    return _updateAgentDeserialize(result);
}
export function _getAgentSend(context, assistantId, options = { requestOptions: {} }) {
    var _a, _b;
    const path = expandUrlTemplate("/assistants/{assistantId}{?api%2Dversion}", {
        assistantId: assistantId,
        "api%2Dversion": context.apiVersion,
    }, {
        allowReserved: (_a = options === null || options === void 0 ? void 0 : options.requestOptions) === null || _a === void 0 ? void 0 : _a.skipUrlEncoding,
    });
    return context.path(path).get(Object.assign(Object.assign({}, operationOptionsToRequestParameters(options)), { headers: Object.assign({ accept: "application/json" }, (_b = options.requestOptions) === null || _b === void 0 ? void 0 : _b.headers) }));
}
export async function _getAgentDeserialize(result) {
    const expectedStatuses = ["200"];
    if (!expectedStatuses.includes(result.status)) {
        throw createRestError(result);
    }
    return agentDeserializer(result.body);
}
/** Retrieves an existing agent. */
export async function getAgent(context, assistantId, options = { requestOptions: {} }) {
    const result = await _getAgentSend(context, assistantId, options);
    return _getAgentDeserialize(result);
}
export function _listAgentsSend(context, options = { requestOptions: {} }) {
    var _a, _b;
    const path = expandUrlTemplate("/assistants{?api%2Dversion,limit,order,after,before}", {
        "api%2Dversion": context.apiVersion,
        limit: options === null || options === void 0 ? void 0 : options.limit,
        order: options === null || options === void 0 ? void 0 : options.order,
        after: options === null || options === void 0 ? void 0 : options.after,
        before: options === null || options === void 0 ? void 0 : options.before,
    }, {
        allowReserved: (_a = options === null || options === void 0 ? void 0 : options.requestOptions) === null || _a === void 0 ? void 0 : _a.skipUrlEncoding,
    });
    return context.path(path).get(Object.assign(Object.assign({}, operationOptionsToRequestParameters(options)), { headers: Object.assign({ accept: "application/json" }, (_b = options.requestOptions) === null || _b === void 0 ? void 0 : _b.headers) }));
}
export async function _listAgentsDeserialize(result) {
    const expectedStatuses = ["200"];
    if (!expectedStatuses.includes(result.status)) {
        throw createRestError(result);
    }
    return _agentsPagedResultAgentDeserializer(result.body);
}
/** Gets a list of agents that were previously created. */
export function listAgents(context, options = { requestOptions: {} }) {
    return buildPagedAsyncIterator(context, () => _listAgentsSend(context, options), _listAgentsDeserialize, ["200"], { itemName: "data" });
}
export function _createAgentSend(context, model, options = { requestOptions: {} }) {
    var _a, _b;
    const path = expandUrlTemplate("/assistants{?api%2Dversion}", {
        "api%2Dversion": context.apiVersion,
    }, {
        allowReserved: (_a = options === null || options === void 0 ? void 0 : options.requestOptions) === null || _a === void 0 ? void 0 : _a.skipUrlEncoding,
    });
    return context.path(path).post(Object.assign(Object.assign({}, operationOptionsToRequestParameters(options)), { contentType: "application/json", headers: Object.assign({ accept: "application/json" }, (_b = options.requestOptions) === null || _b === void 0 ? void 0 : _b.headers), body: {
            model: model,
            name: options === null || options === void 0 ? void 0 : options.name,
            description: options === null || options === void 0 ? void 0 : options.description,
            instructions: options === null || options === void 0 ? void 0 : options.instructions,
            tools: !(options === null || options === void 0 ? void 0 : options.tools) ? options === null || options === void 0 ? void 0 : options.tools : toolDefinitionUnionArraySerializer(options === null || options === void 0 ? void 0 : options.tools),
            tool_resources: !(options === null || options === void 0 ? void 0 : options.toolResources)
                ? options === null || options === void 0 ? void 0 : options.toolResources
                : toolResourcesSerializer(options === null || options === void 0 ? void 0 : options.toolResources),
            temperature: options === null || options === void 0 ? void 0 : options.temperature,
            top_p: options === null || options === void 0 ? void 0 : options.topP,
            response_format: !(options === null || options === void 0 ? void 0 : options.responseFormat)
                ? options === null || options === void 0 ? void 0 : options.responseFormat
                : agentsResponseFormatOptionSerializer(options === null || options === void 0 ? void 0 : options.responseFormat),
            metadata: options === null || options === void 0 ? void 0 : options.metadata,
        } }));
}
export async function _createAgentDeserialize(result) {
    const expectedStatuses = ["200"];
    if (!expectedStatuses.includes(result.status)) {
        throw createRestError(result);
    }
    return agentDeserializer(result.body);
}
/** Creates a new agent. */
export async function createAgent(context, model, options = { requestOptions: {} }) {
    const result = await _createAgentSend(context, model, options);
    return _createAgentDeserialize(result);
}
const handlers = [
    { events: Object.values(ThreadStreamEvent) },
    { events: Object.values(RunStreamEvent) },
    { events: Object.values(RunStepStreamEvent) },
    { events: Object.values(MessageStreamEvent) },
];
function createAgentStream(stream) {
    const asyncIterator = toAsyncIterable(stream);
    const asyncDisposable = stream;
    return Object.assign(asyncIterator, asyncDisposable);
}
function toAsyncIterable(stream) {
    return __asyncGenerator(this, arguments, function* toAsyncIterable_1() {
        var _a, e_1, _b, _c;
        try {
            for (var _d = true, stream_1 = __asyncValues(stream), stream_1_1; stream_1_1 = yield __await(stream_1.next()), _a = stream_1_1.done, !_a; _d = true) {
                _c = stream_1_1.value;
                _d = false;
                const event = _c;
                const data = deserializeEventData(event);
                yield yield __await({ data: data, event: event.event });
            }
        }
        catch (e_1_1) { e_1 = { error: e_1_1 }; }
        finally {
            try {
                if (!_d && !_a && (_b = stream_1.return)) yield __await(_b.call(stream_1));
            }
            finally { if (e_1) throw e_1.error; }
        }
    });
}
function deserializeEventData(event) {
    try {
        const jsonData = JSON.parse(event.data);
        switch (event.event) {
            case MessageStreamEvent.ThreadMessageDelta:
                return jsonData;
            case RunStepStreamEvent.ThreadRunStepDelta:
                return jsonData;
            default: {
                for (const { events } of handlers) {
                    if (events.includes(event.event)) {
                        return jsonData;
                    }
                }
                return jsonData;
            }
        }
    }
    catch (ex) {
        logger.error(`Failed to parse event data  ${event.event} - error: ${ex}`);
        return event.data;
    }
}
async function processStream(streamResponse) {
    const expectedStatuses = ["200"];
    const result = isNodeLike
        ? await streamResponse.asNodeStream()
        : await streamResponse.asBrowserStream();
    if (!expectedStatuses.includes(result.status)) {
        throw createRestError(result);
    }
    if (!result.body) {
        throw new Error("No body in response");
    }
    const stream = isNodeLike
        ? createSseStream(result.body)
        : createSseStream(result.body);
    return createAgentStream(stream);
}
/** Create a run and stream the events */
export async function createRunStreaming(context, assistantId, threadId, options = { requestOptions: {} }) {
    const streamOptions = Object.assign(Object.assign({}, options), { stream: true });
    return processStream(_createRunSend(context, threadId, assistantId, streamOptions));
}
/** Create a thread and run and stream the events */
export async function createThreadAndRunStreaming(context, assistantId, options = { requestOptions: {} }) {
    const streamOptions = Object.assign(Object.assign({}, options), { stream: true });
    return processStream(_createThreadAndRunSend(context, assistantId, streamOptions));
}
export async function submitToolOutputsToRunStreaming(context, threadId, runId, options = { requestOptions: {} }) {
    const streamOptions = Object.assign(Object.assign({}, options), { stream: true });
    return processStream(context
        .path("/threads/{threadId}/runs/{runId}/submit_tool_outputs", threadId, runId)
        .post(streamOptions));
}
