// @azure/ai-projects/dist/browser/classic/datasets/index.js
// Copyright (c) Microsoft Corporation.
// Licensed under the MIT License.
/* eslint-disable tsdoc/syntax */
import { getCredentials, pendingUpload, createOrUpdate, $delete, get, list, listVersions, uploadFile, uploadFolder, } from "../../api/datasets/operations.js";
function _getDatasets(context) {
    return {
        getCredentials: (name, version, options) => getCredentials(context, name, version, options),
        pendingUpload: (name, version, body, options) => pendingUpload(context, name, version, body, options),
        createOrUpdate: (name, version, body, options) => createOrUpdate(context, name, version, body, options),
        delete: (name, version, options) => $delete(context, name, version, options),
        get: (name, version, options) => get(context, name, version, options),
        list: (options) => list(context, options),
        listVersions: (name, options) => listVersions(context, name, options),
        uploadFile: (name, version, filePath, connectionName) => uploadFile(context, name, version, filePath, connectionName),
        uploadFolder: (name, version, folderPath, connectionName) => uploadFolder(context, name, version, folderPath, connectionName),
    };
}
export function _getDatasetsOperations(context) {
    return Object.assign({}, _getDatasets(context));
}
