// @azure/ai-agents/dist/commonjs/classic/runs/index.js
"use strict";
// Copyright (c) Microsoft Corporation.
// Licensed under the MIT License.
Object.defineProperty(exports, "__esModule", { value: true });
exports._getRunsOperations = _getRunsOperations;
const operations_js_1 = require("../../api/runs/operations.js");
const operations_js_2 = require("../../api/operations.js");
function _getRuns(context) {
    return {
        cancel: (threadId, runId, options) => (0, operations_js_1.cancelRun)(context, threadId, runId, options),
        submitToolOutputs: (threadId, runId, toolOutputs, options) => (0, operations_js_1.submitToolOutputsToRun)(context, threadId, runId, toolOutputs, options),
        update: (threadId, runId, options) => (0, operations_js_1.updateRun)(context, threadId, runId, options),
        get: (threadId, runId, options) => (0, operations_js_1.getRun)(context, threadId, runId, options),
        list: (threadId, options) => (0, operations_js_1.listRuns)(context, threadId, options),
        create: (threadId, assistantId, options) => (0, operations_js_1.createRun)(context, threadId, assistantId, options),
        createAndPoll: (threadId, assistantId, options) => (0, operations_js_1.createRunAndPoll)(context, threadId, assistantId, options),
        createThreadAndRun: (assistantId, options) => (0, operations_js_2.createThreadAndRun)(context, assistantId, options),
    };
}
function _getRunsOperations(context) {
    return Object.assign({}, _getRuns(context));
}
