// @azure/ai-agents/dist/commonjs/api/messages/operations.js
"use strict";
// Copyright (c) Microsoft Corporation.
// Licensed under the MIT License.
Object.defineProperty(exports, "__esModule", { value: true });
exports._updateMessageSend = _updateMessageSend;
exports._updateMessageDeserialize = _updateMessageDeserialize;
exports.updateMessage = updateMessage;
exports._getMessageSend = _getMessageSend;
exports._getMessageDeserialize = _getMessageDeserialize;
exports.getMessage = getMessage;
exports._listMessagesSend = _listMessagesSend;
exports._listMessagesDeserialize = _listMessagesDeserialize;
exports.listMessages = listMessages;
exports._createMessageSend = _createMessageSend;
exports._createMessageDeserialize = _createMessageDeserialize;
exports.createMessage = createMessage;
const models_js_1 = require("../../models/models.js");
const pagingHelpers_js_1 = require("../../static-helpers/pagingHelpers.js");
const urlTemplate_js_1 = require("../../static-helpers/urlTemplate.js");
const core_client_1 = require("@azure-rest/core-client");
function _updateMessageSend(context, threadId, messageId, options = { requestOptions: {} }) {
    var _a, _b;
    const path = (0, urlTemplate_js_1.expandUrlTemplate)("/threads/{threadId}/messages/{messageId}{?api%2Dversion}", {
        threadId: threadId,
        messageId: messageId,
        "api%2Dversion": context.apiVersion,
    }, {
        allowReserved: (_a = options === null || options === void 0 ? void 0 : options.requestOptions) === null || _a === void 0 ? void 0 : _a.skipUrlEncoding,
    });
    return context.path(path).post(Object.assign(Object.assign({}, (0, core_client_1.operationOptionsToRequestParameters)(options)), { contentType: "application/json", headers: Object.assign({ accept: "application/json" }, (_b = options.requestOptions) === null || _b === void 0 ? void 0 : _b.headers), body: { metadata: options === null || options === void 0 ? void 0 : options.metadata } }));
}
async function _updateMessageDeserialize(result) {
    const expectedStatuses = ["200"];
    if (!expectedStatuses.includes(result.status)) {
        throw (0, core_client_1.createRestError)(result);
    }
    return (0, models_js_1.threadMessageDeserializer)(result.body);
}
/** Modifies an existing message on an existing thread. */
async function updateMessage(context, threadId, messageId, options = { requestOptions: {} }) {
    const result = await _updateMessageSend(context, threadId, messageId, options);
    return _updateMessageDeserialize(result);
}
function _getMessageSend(context, threadId, messageId, options = { requestOptions: {} }) {
    var _a, _b;
    const path = (0, urlTemplate_js_1.expandUrlTemplate)("/threads/{threadId}/messages/{messageId}{?api%2Dversion}", {
        threadId: threadId,
        messageId: messageId,
        "api%2Dversion": context.apiVersion,
    }, {
        allowReserved: (_a = options === null || options === void 0 ? void 0 : options.requestOptions) === null || _a === void 0 ? void 0 : _a.skipUrlEncoding,
    });
    return context.path(path).get(Object.assign(Object.assign({}, (0, core_client_1.operationOptionsToRequestParameters)(options)), { headers: Object.assign({ accept: "application/json" }, (_b = options.requestOptions) === null || _b === void 0 ? void 0 : _b.headers) }));
}
async function _getMessageDeserialize(result) {
    const expectedStatuses = ["200"];
    if (!expectedStatuses.includes(result.status)) {
        throw (0, core_client_1.createRestError)(result);
    }
    return (0, models_js_1.threadMessageDeserializer)(result.body);
}
/** Retrieves an existing message. */
async function getMessage(context, threadId, messageId, options = { requestOptions: {} }) {
    const result = await _getMessageSend(context, threadId, messageId, options);
    return _getMessageDeserialize(result);
}
function _listMessagesSend(context, threadId, options = { requestOptions: {} }) {
    var _a, _b;
    const path = (0, urlTemplate_js_1.expandUrlTemplate)("/threads/{threadId}/messages{?run_id,api%2Dversion,limit,order,after,before}", {
        threadId: threadId,
        run_id: options === null || options === void 0 ? void 0 : options.runId,
        "api%2Dversion": context.apiVersion,
        limit: options === null || options === void 0 ? void 0 : options.limit,
        order: options === null || options === void 0 ? void 0 : options.order,
        after: options === null || options === void 0 ? void 0 : options.after,
        before: options === null || options === void 0 ? void 0 : options.before,
    }, {
        allowReserved: (_a = options === null || options === void 0 ? void 0 : options.requestOptions) === null || _a === void 0 ? void 0 : _a.skipUrlEncoding,
    });
    return context.path(path).get(Object.assign(Object.assign({}, (0, core_client_1.operationOptionsToRequestParameters)(options)), { headers: Object.assign({ accept: "application/json" }, (_b = options.requestOptions) === null || _b === void 0 ? void 0 : _b.headers) }));
}
async function _listMessagesDeserialize(result) {
    const expectedStatuses = ["200"];
    if (!expectedStatuses.includes(result.status)) {
        throw (0, core_client_1.createRestError)(result);
    }
    return (0, models_js_1._agentsPagedResultThreadMessageDeserializer)(result.body);
}
/** Gets a list of messages that exist on a thread. */
function listMessages(context, threadId, options = { requestOptions: {} }) {
    return (0, pagingHelpers_js_1.buildPagedAsyncIterator)(context, () => _listMessagesSend(context, threadId, options), _listMessagesDeserialize, ["200"], { itemName: "data" });
}
function _createMessageSend(context, threadId, role, content, options = { requestOptions: {} }) {
    var _a, _b;
    const path = (0, urlTemplate_js_1.expandUrlTemplate)("/threads/{threadId}/messages{?api%2Dversion}", {
        threadId: threadId,
        "api%2Dversion": context.apiVersion,
    }, {
        allowReserved: (_a = options === null || options === void 0 ? void 0 : options.requestOptions) === null || _a === void 0 ? void 0 : _a.skipUrlEncoding,
    });
    return context.path(path).post(Object.assign(Object.assign({}, (0, core_client_1.operationOptionsToRequestParameters)(options)), { contentType: "application/json", headers: Object.assign({ accept: "application/json" }, (_b = options.requestOptions) === null || _b === void 0 ? void 0 : _b.headers), body: {
            role: role,
            content: (0, models_js_1.messageInputContentSerializer)(content),
            attachments: !(options === null || options === void 0 ? void 0 : options.attachments)
                ? options === null || options === void 0 ? void 0 : options.attachments
                : (0, models_js_1.messageAttachmentArraySerializer)(options === null || options === void 0 ? void 0 : options.attachments),
            metadata: options === null || options === void 0 ? void 0 : options.metadata,
        } }));
}
async function _createMessageDeserialize(result) {
    const expectedStatuses = ["200"];
    if (!expectedStatuses.includes(result.status)) {
        throw (0, core_client_1.createRestError)(result);
    }
    return (0, models_js_1.threadMessageDeserializer)(result.body);
}
/** Creates a new message on a specified thread. */
async function createMessage(context, threadId, role, content, options = { requestOptions: {} }) {
    const result = await _createMessageSend(context, threadId, role, content, options);
    return _createMessageDeserialize(result);
}
