// @azure/ai-agents/dist/react-native/api/files/operations.js
// Copyright (c) Microsoft Corporation.
// Licensed under the MIT License.
import { fileListResponseDeserializer, fileInfoDeserializer, _uploadFileRequestSerializer, fileDeletionStatusDeserializer, } from "../../models/models.js";
import { expandUrlTemplate } from "../../static-helpers/urlTemplate.js";
import { createRestError, operationOptionsToRequestParameters, } from "@azure-rest/core-client";
import { createPoller } from "../poller.js";
export function _getFileContentSend(context, fileId, options = { requestOptions: {} }) {
    var _a, _b;
    const path = expandUrlTemplate("/files/{fileId}/content{?api%2Dversion}", {
        fileId: fileId,
        "api%2Dversion": context.apiVersion,
    }, {
        allowReserved: (_a = options === null || options === void 0 ? void 0 : options.requestOptions) === null || _a === void 0 ? void 0 : _a.skipUrlEncoding,
    });
    return context.path(path).get(Object.assign(Object.assign({}, operationOptionsToRequestParameters(options)), { headers: Object.assign({ accept: "application/octet-stream" }, (_b = options.requestOptions) === null || _b === void 0 ? void 0 : _b.headers) }));
}
/** Retrieves the raw content of a specific file. */
export function getFileContent(context, fileId, options = { requestOptions: {} }) {
    const result = _getFileContentSend(context, fileId, options);
    return result;
}
export function _getFileSend(context, fileId, options = { requestOptions: {} }) {
    var _a, _b;
    const path = expandUrlTemplate("/files/{fileId}{?api%2Dversion}", {
        fileId: fileId,
        "api%2Dversion": context.apiVersion,
    }, {
        allowReserved: (_a = options === null || options === void 0 ? void 0 : options.requestOptions) === null || _a === void 0 ? void 0 : _a.skipUrlEncoding,
    });
    return context.path(path).get(Object.assign(Object.assign({}, operationOptionsToRequestParameters(options)), { headers: Object.assign({ accept: "application/json" }, (_b = options.requestOptions) === null || _b === void 0 ? void 0 : _b.headers) }));
}
export async function _getFileDeserialize(result) {
    const expectedStatuses = ["200"];
    if (!expectedStatuses.includes(result.status)) {
        throw createRestError(result);
    }
    return fileInfoDeserializer(result.body);
}
/** Returns information about a specific file. Does not retrieve file content. */
export async function getFile(context, fileId, options = { requestOptions: {} }) {
    const result = await _getFileSend(context, fileId, options);
    return _getFileDeserialize(result);
}
export function _deleteFileSend(context, fileId, options = { requestOptions: {} }) {
    var _a, _b;
    const path = expandUrlTemplate("/files/{fileId}{?api%2Dversion}", {
        fileId: fileId,
        "api%2Dversion": context.apiVersion,
    }, {
        allowReserved: (_a = options === null || options === void 0 ? void 0 : options.requestOptions) === null || _a === void 0 ? void 0 : _a.skipUrlEncoding,
    });
    return context.path(path).delete(Object.assign(Object.assign({}, operationOptionsToRequestParameters(options)), { headers: Object.assign({ accept: "application/json" }, (_b = options.requestOptions) === null || _b === void 0 ? void 0 : _b.headers) }));
}
export async function _deleteFileDeserialize(result) {
    const expectedStatuses = ["200"];
    if (!expectedStatuses.includes(result.status)) {
        throw createRestError(result);
    }
    return fileDeletionStatusDeserializer(result.body);
}
/** Delete a previously uploaded file. */
export async function deleteFile(context, fileId, options = { requestOptions: {} }) {
    const result = await _deleteFileSend(context, fileId, options);
    return _deleteFileDeserialize(result);
}
export function _uploadFileSend(context, body, options = { requestOptions: {} }) {
    var _a;
    return context.path("/files").post(Object.assign(Object.assign({}, operationOptionsToRequestParameters(options)), { contentType: "multipart/form-data", headers: Object.assign({ accept: "application/json" }, (_a = options.requestOptions) === null || _a === void 0 ? void 0 : _a.headers), body: _uploadFileRequestSerializer(body) }));
}
export async function _uploadFileDeserialize(result) {
    const expectedStatuses = ["200"];
    if (!expectedStatuses.includes(result.status)) {
        throw createRestError(result);
    }
    return fileInfoDeserializer(result.body);
}
/** Uploads a file for use by other operations. */
export async function uploadFileInternal(context, body, options = { requestOptions: {} }) {
    const result = await _uploadFileSend(context, body, options);
    return _uploadFileDeserialize(result);
}
/** Uploads a file for use by other operations. */
export function uploadFile(context, body, options = { requestOptions: {} }) {
    var _a;
    return createPoller({
        initOperation: async () => {
            return uploadFileInternal(context, body, options);
        },
        pollOperation: async (currentResult) => {
            return getFile(context, currentResult.id, options);
        },
        getOperationStatus: getLroOperationStatus,
        getOperationError: (result) => {
            return getLroOperationStatus(result) === "failed" && result.statusDetails
                ? new Error(`Operation failed: ${result.statusDetails}`)
                : undefined;
        },
        intervalInMs: (_a = options.pollingOptions) === null || _a === void 0 ? void 0 : _a.intervalInMs,
    });
}
/** Uploads a file for use by other operations with polling */
export function uploadFileAndPoll(context, body, options = { requestOptions: {} }) {
    var _a;
    return createPoller({
        initOperation: async () => {
            return uploadFileInternal(context, body, options);
        },
        pollOperation: async (currentResult) => {
            return getFile(context, currentResult.id, options);
        },
        getOperationStatus: getLroOperationStatus,
        getOperationError: (result) => {
            return getLroOperationStatus(result) === "failed" && result.statusDetails
                ? new Error(`Operation failed: ${result.statusDetails}`)
                : undefined;
        },
        intervalInMs: (_a = options.pollingOptions) === null || _a === void 0 ? void 0 : _a.intervalInMs,
    });
}
export function _listFilesSend(context, options = { requestOptions: {} }) {
    var _a, _b;
    const path = expandUrlTemplate("/files{?api%2Dversion,purpose}", {
        "api%2Dversion": context.apiVersion,
        purpose: options === null || options === void 0 ? void 0 : options.purpose,
    }, {
        allowReserved: (_a = options === null || options === void 0 ? void 0 : options.requestOptions) === null || _a === void 0 ? void 0 : _a.skipUrlEncoding,
    });
    return context.path(path).get(Object.assign(Object.assign({}, operationOptionsToRequestParameters(options)), { headers: Object.assign({ accept: "application/json" }, (_b = options.requestOptions) === null || _b === void 0 ? void 0 : _b.headers) }));
}
export async function _listFilesDeserialize(result) {
    const expectedStatuses = ["200"];
    if (!expectedStatuses.includes(result.status)) {
        throw createRestError(result);
    }
    return fileListResponseDeserializer(result.body);
}
/** Gets a list of previously uploaded files. */
export async function listFiles(context, options = { requestOptions: {} }) {
    const result = await _listFilesSend(context, options);
    return _listFilesDeserialize(result);
}
function getLroOperationStatus(result) {
    switch (result.status) {
        case "running":
        case "pending":
            return "running";
        case "uploaded":
        case "processed":
            return "succeeded";
        default:
            return "failed";
    }
}
