// @azure/ai-projects/dist/commonjs/models/models.js
"use strict";
// Copyright (c) Microsoft Corporation.
// Licensed under the MIT License.
/* eslint-disable @typescript-eslint/explicit-module-boundary-types */
/* eslint-disable @typescript-eslint/naming-convention */
Object.defineProperty(exports, "__esModule", { value: true });
exports.KnownVersions = void 0;
exports.connectionDeserializer = connectionDeserializer;
exports.baseCredentialsDeserializer = baseCredentialsDeserializer;
exports.baseCredentialsUnionDeserializer = baseCredentialsUnionDeserializer;
exports.apiKeyCredentialsDeserializer = apiKeyCredentialsDeserializer;
exports.entraIDCredentialsDeserializer = entraIDCredentialsDeserializer;
exports.customCredentialDeserializer = customCredentialDeserializer;
exports.sasCredentialsDeserializer = sasCredentialsDeserializer;
exports.noAuthenticationCredentialsDeserializer = noAuthenticationCredentialsDeserializer;
exports._pagedConnectionDeserializer = _pagedConnectionDeserializer;
exports.connectionArrayDeserializer = connectionArrayDeserializer;
exports.evaluationSerializer = evaluationSerializer;
exports.evaluationDeserializer = evaluationDeserializer;
exports.inputDataSerializer = inputDataSerializer;
exports.inputDataDeserializer = inputDataDeserializer;
exports.inputDataUnionSerializer = inputDataUnionSerializer;
exports.inputDataUnionDeserializer = inputDataUnionDeserializer;
exports.inputDatasetSerializer = inputDatasetSerializer;
exports.inputDatasetDeserializer = inputDatasetDeserializer;
exports.evaluatorConfigurationRecordSerializer = evaluatorConfigurationRecordSerializer;
exports.evaluatorConfigurationRecordDeserializer = evaluatorConfigurationRecordDeserializer;
exports.evaluatorConfigurationSerializer = evaluatorConfigurationSerializer;
exports.evaluatorConfigurationDeserializer = evaluatorConfigurationDeserializer;
exports._pagedEvaluationDeserializer = _pagedEvaluationDeserializer;
exports.evaluationArraySerializer = evaluationArraySerializer;
exports.evaluationArrayDeserializer = evaluationArrayDeserializer;
exports.agentEvaluationRequestSerializer = agentEvaluationRequestSerializer;
exports.agentEvaluationSamplingConfigurationSerializer = agentEvaluationSamplingConfigurationSerializer;
exports.agentEvaluationRedactionConfigurationSerializer = agentEvaluationRedactionConfigurationSerializer;
exports.agentEvaluationDeserializer = agentEvaluationDeserializer;
exports.agentEvaluationResultArrayDeserializer = agentEvaluationResultArrayDeserializer;
exports.agentEvaluationResultDeserializer = agentEvaluationResultDeserializer;
exports._pagedDatasetVersionDeserializer = _pagedDatasetVersionDeserializer;
exports.datasetVersionUnionArraySerializer = datasetVersionUnionArraySerializer;
exports.datasetVersionUnionArrayDeserializer = datasetVersionUnionArrayDeserializer;
exports.datasetVersionSerializer = datasetVersionSerializer;
exports.datasetVersionDeserializer = datasetVersionDeserializer;
exports.datasetVersionUnionSerializer = datasetVersionUnionSerializer;
exports.datasetVersionUnionDeserializer = datasetVersionUnionDeserializer;
exports.fileDatasetVersionSerializer = fileDatasetVersionSerializer;
exports.fileDatasetVersionDeserializer = fileDatasetVersionDeserializer;
exports.folderDatasetVersionSerializer = folderDatasetVersionSerializer;
exports.folderDatasetVersionDeserializer = folderDatasetVersionDeserializer;
exports.pendingUploadRequestSerializer = pendingUploadRequestSerializer;
exports.pendingUploadResponseDeserializer = pendingUploadResponseDeserializer;
exports.blobReferenceDeserializer = blobReferenceDeserializer;
exports.sasCredentialDeserializer = sasCredentialDeserializer;
exports.assetCredentialResponseDeserializer = assetCredentialResponseDeserializer;
exports._pagedIndexDeserializer = _pagedIndexDeserializer;
exports.indexUnionArraySerializer = indexUnionArraySerializer;
exports.indexUnionArrayDeserializer = indexUnionArrayDeserializer;
exports.indexSerializer = indexSerializer;
exports.indexDeserializer = indexDeserializer;
exports.indexUnionSerializer = indexUnionSerializer;
exports.indexUnionDeserializer = indexUnionDeserializer;
exports.azureAISearchIndexSerializer = azureAISearchIndexSerializer;
exports.azureAISearchIndexDeserializer = azureAISearchIndexDeserializer;
exports.fieldMappingSerializer = fieldMappingSerializer;
exports.fieldMappingDeserializer = fieldMappingDeserializer;
exports.managedAzureAISearchIndexSerializer = managedAzureAISearchIndexSerializer;
exports.managedAzureAISearchIndexDeserializer = managedAzureAISearchIndexDeserializer;
exports.cosmosDBIndexSerializer = cosmosDBIndexSerializer;
exports.cosmosDBIndexDeserializer = cosmosDBIndexDeserializer;
exports.embeddingConfigurationSerializer = embeddingConfigurationSerializer;
exports.embeddingConfigurationDeserializer = embeddingConfigurationDeserializer;
exports.deploymentDeserializer = deploymentDeserializer;
exports.deploymentUnionDeserializer = deploymentUnionDeserializer;
exports.modelDeploymentDeserializer = modelDeploymentDeserializer;
exports.skuDeserializer = skuDeserializer;
exports._pagedDeploymentDeserializer = _pagedDeploymentDeserializer;
exports.deploymentUnionArrayDeserializer = deploymentUnionArrayDeserializer;
exports.redTeamSerializer = redTeamSerializer;
exports.redTeamDeserializer = redTeamDeserializer;
exports.targetConfigSerializer = targetConfigSerializer;
exports.targetConfigDeserializer = targetConfigDeserializer;
exports.targetConfigUnionSerializer = targetConfigUnionSerializer;
exports.targetConfigUnionDeserializer = targetConfigUnionDeserializer;
exports.azureOpenAIModelConfigurationSerializer = azureOpenAIModelConfigurationSerializer;
exports.azureOpenAIModelConfigurationDeserializer = azureOpenAIModelConfigurationDeserializer;
exports._pagedRedTeamDeserializer = _pagedRedTeamDeserializer;
exports.redTeamArraySerializer = redTeamArraySerializer;
exports.redTeamArrayDeserializer = redTeamArrayDeserializer;
function connectionDeserializer(item) {
    return {
        name: item["name"],
        id: item["id"],
        type: item["type"],
        target: item["target"],
        metadata: item["metadata"],
        isDefault: item["isDefault"],
        credentials: item["credentials"]
            ? baseCredentialsUnionDeserializer(item["credentials"])
            : { type: "None" },
    };
}
function baseCredentialsDeserializer(item) {
    return {
        type: item["type"],
    };
}
function baseCredentialsUnionDeserializer(item) {
    switch (item.type) {
        case "ApiKey":
            return apiKeyCredentialsDeserializer(item);
        case "AAD":
            return entraIDCredentialsDeserializer(item);
        case "CustomKeys":
            return customCredentialDeserializer(item);
        case "SAS":
            return sasCredentialsDeserializer(item);
        case "None":
            return noAuthenticationCredentialsDeserializer(item);
        default:
            return baseCredentialsDeserializer(item);
    }
}
function apiKeyCredentialsDeserializer(item) {
    return {
        type: item["type"],
        apiKey: item["key"],
    };
}
function entraIDCredentialsDeserializer(item) {
    return {
        type: item["type"],
    };
}
function customCredentialDeserializer(item) {
    return {
        type: item["type"],
        keys: item["keys"],
    };
}
function sasCredentialsDeserializer(item) {
    return {
        type: item["type"],
        sasToken: item["SAS"],
    };
}
function noAuthenticationCredentialsDeserializer(item) {
    return {
        type: item["type"],
    };
}
function _pagedConnectionDeserializer(item) {
    return {
        value: connectionArrayDeserializer(item["value"]),
        nextLink: item["nextLink"],
    };
}
function connectionArrayDeserializer(result) {
    return result.map((item) => {
        return connectionDeserializer(item);
    });
}
// optional name property
function evaluationSerializer(item) {
    return {
        data: inputDataUnionSerializer(item["data"]),
        displayName: item["displayName"],
        description: item["description"],
        tags: item["tags"],
        properties: item["properties"],
        evaluators: evaluatorConfigurationRecordSerializer(item["evaluators"]),
    };
}
function evaluationDeserializer(item) {
    return {
        name: item["id"],
        data: inputDataUnionDeserializer(item["data"]),
        displayName: item["displayName"],
        description: item["description"],
        status: item["status"],
        tags: item["tags"],
        properties: item["properties"],
        evaluators: evaluatorConfigurationRecordDeserializer(item["evaluators"]),
    };
}
function inputDataSerializer(item) {
    return { type: item["type"] };
}
function inputDataDeserializer(item) {
    return {
        type: item["type"],
    };
}
function inputDataUnionSerializer(item) {
    switch (item.type) {
        case "dataset":
            return inputDatasetSerializer(item);
        default:
            return inputDataSerializer(item);
    }
}
function inputDataUnionDeserializer(_item) {
    const item = _item || {};
    switch (item.type) {
        case "dataset":
            return inputDatasetDeserializer(item);
        default:
            return inputDataDeserializer(item);
    }
}
function inputDatasetSerializer(item) {
    return { type: item["type"], id: item["id"] };
}
function inputDatasetDeserializer(item) {
    return {
        type: item["type"],
        id: item["id"],
    };
}
function evaluatorConfigurationRecordSerializer(item) {
    const result = {};
    Object.keys(item).map((key) => {
        result[key] = !item[key] ? item[key] : evaluatorConfigurationSerializer(item[key]);
    });
    return result;
}
function evaluatorConfigurationRecordDeserializer(item) {
    const result = {};
    Object.keys(item).map((key) => {
        result[key] = !item[key] ? item[key] : evaluatorConfigurationDeserializer(item[key]);
    });
    return result;
}
function evaluatorConfigurationSerializer(item) {
    return {
        id: item["id"],
        initParams: item["initParams"],
        dataMapping: item["dataMapping"],
    };
}
function evaluatorConfigurationDeserializer(item) {
    return {
        id: item["id"],
        initParams: item["initParams"],
        dataMapping: item["dataMapping"],
    };
}
function _pagedEvaluationDeserializer(item) {
    return {
        value: evaluationArrayDeserializer(item["value"]),
        nextLink: item["nextLink"],
    };
}
function evaluationArraySerializer(result) {
    return result.map((item) => {
        return evaluationSerializer(item);
    });
}
function evaluationArrayDeserializer(result) {
    return result.map((item) => {
        return evaluationDeserializer(item);
    });
}
function agentEvaluationRequestSerializer(item) {
    return {
        runId: item["runId"],
        threadId: item["threadId"],
        evaluators: evaluatorConfigurationRecordSerializer(item["evaluators"]),
        samplingConfiguration: !item["samplingConfiguration"]
            ? item["samplingConfiguration"]
            : agentEvaluationSamplingConfigurationSerializer(item["samplingConfiguration"]),
        redactionConfiguration: !item["redactionConfiguration"]
            ? item["redactionConfiguration"]
            : agentEvaluationRedactionConfigurationSerializer(item["redactionConfiguration"]),
        appInsightsConnectionString: item["appInsightsConnectionString"],
    };
}
function agentEvaluationSamplingConfigurationSerializer(item) {
    return {
        name: item["name"],
        samplingPercent: item["samplingPercent"],
        maxRequestRate: item["maxRequestRate"],
    };
}
function agentEvaluationRedactionConfigurationSerializer(item) {
    return { redactScoreProperties: item["redactScoreProperties"] };
}
function agentEvaluationDeserializer(item) {
    return {
        id: item["id"],
        status: item["status"],
        error: item["error"],
        result: !item["result"]
            ? item["result"]
            : agentEvaluationResultArrayDeserializer(item["result"]),
    };
}
function agentEvaluationResultArrayDeserializer(result) {
    return result.map((item) => {
        return agentEvaluationResultDeserializer(item);
    });
}
function agentEvaluationResultDeserializer(item) {
    return {
        evaluator: item["evaluator"],
        evaluatorId: item["evaluatorId"],
        score: item["score"],
        status: item["status"],
        reason: item["reason"],
        version: item["version"],
        threadId: item["threadId"],
        runId: item["runId"],
        error: item["error"],
        additionalDetails: item["additionalDetails"],
    };
}
function _pagedDatasetVersionDeserializer(item) {
    return {
        value: datasetVersionUnionArrayDeserializer(item["value"]),
        nextLink: item["nextLink"],
    };
}
function datasetVersionUnionArraySerializer(result) {
    return result.map((item) => {
        return datasetVersionUnionSerializer(item);
    });
}
function datasetVersionUnionArrayDeserializer(result) {
    return result.map((item) => {
        return datasetVersionUnionDeserializer(item);
    });
}
function datasetVersionSerializer(item) {
    return {
        dataUri: item["dataUri"],
        type: item["type"],
        connectionName: item["connectionName"],
        description: item["description"],
        tags: item["tags"],
    };
}
function datasetVersionDeserializer(item) {
    return {
        dataUri: item["dataUri"],
        type: item["type"],
        isReference: item["isReference"],
        connectionName: item["connectionName"],
        id: item["id"],
        name: item["name"],
        version: item["version"],
        description: item["description"],
        tags: item["tags"],
    };
}
function datasetVersionUnionSerializer(item) {
    switch (item.type) {
        case "uri_file":
            return fileDatasetVersionSerializer(item);
        case "uri_folder":
            return folderDatasetVersionSerializer(item);
        default:
            return datasetVersionSerializer(item);
    }
}
function datasetVersionUnionDeserializer(item) {
    switch (item.type) {
        case "uri_file":
            return fileDatasetVersionDeserializer(item);
        case "uri_folder":
            return folderDatasetVersionDeserializer(item);
        default:
            return datasetVersionDeserializer(item);
    }
}
function fileDatasetVersionSerializer(item) {
    return {
        dataUri: item["dataUri"],
        type: item["type"],
        connectionName: item["connectionName"],
        description: item["description"],
        tags: item["tags"],
    };
}
function fileDatasetVersionDeserializer(item) {
    return {
        dataUri: item["dataUri"],
        type: item["type"],
        isReference: item["isReference"],
        connectionName: item["connectionName"],
        id: item["id"],
        name: item["name"],
        version: item["version"],
        description: item["description"],
        tags: item["tags"],
    };
}
function folderDatasetVersionSerializer(item) {
    return {
        dataUri: item["dataUri"],
        type: item["type"],
        connectionName: item["connectionName"],
        description: item["description"],
        tags: item["tags"],
    };
}
function folderDatasetVersionDeserializer(item) {
    return {
        dataUri: item["dataUri"],
        type: item["type"],
        isReference: item["isReference"],
        connectionName: item["connectionName"],
        id: item["id"],
        name: item["name"],
        version: item["version"],
        description: item["description"],
        tags: item["tags"],
    };
}
function pendingUploadRequestSerializer(item) {
    return {
        pendingUploadId: item["pendingUploadId"],
        connectionName: item["connectionName"],
        pendingUploadType: item["pendingUploadType"],
    };
}
function pendingUploadResponseDeserializer(item) {
    return {
        blobReference: blobReferenceDeserializer(item["blobReference"]),
        pendingUploadId: item["pendingUploadId"],
        version: item["version"],
        pendingUploadType: item["pendingUploadType"],
    };
}
function blobReferenceDeserializer(item) {
    return {
        blobUri: item["blobUri"],
        storageAccountArmId: item["storageAccountArmId"],
        credential: sasCredentialDeserializer(item["credential"]),
    };
}
function sasCredentialDeserializer(item) {
    return {
        sasUri: item["sasUri"],
        type: item["type"],
    };
}
function assetCredentialResponseDeserializer(item) {
    return {
        blobReference: blobReferenceDeserializer(item["blobReference"]),
    };
}
function _pagedIndexDeserializer(item) {
    return {
        value: indexUnionArrayDeserializer(item["value"]),
        nextLink: item["nextLink"],
    };
}
function indexUnionArraySerializer(result) {
    return result.map((item) => {
        return indexUnionSerializer(item);
    });
}
function indexUnionArrayDeserializer(result) {
    return result.map((item) => {
        return indexUnionDeserializer(item);
    });
}
function indexSerializer(item) {
    return {
        type: item["type"],
        description: item["description"],
        tags: item["tags"],
    };
}
function indexDeserializer(item) {
    return {
        type: item["type"],
        id: item["id"],
        name: item["name"],
        version: item["version"],
        description: item["description"],
        tags: item["tags"],
    };
}
function indexUnionSerializer(item) {
    switch (item.type) {
        case "AzureSearch":
            return azureAISearchIndexSerializer(item);
        case "ManagedAzureSearch":
            return managedAzureAISearchIndexSerializer(item);
        case "CosmosDBNoSqlVectorStore":
            return cosmosDBIndexSerializer(item);
        default:
            return indexSerializer(item);
    }
}
function indexUnionDeserializer(item) {
    switch (item.type) {
        case "AzureSearch":
            return azureAISearchIndexDeserializer(item);
        case "ManagedAzureSearch":
            return managedAzureAISearchIndexDeserializer(item);
        case "CosmosDBNoSqlVectorStore":
            return cosmosDBIndexDeserializer(item);
        default:
            return indexDeserializer(item);
    }
}
function azureAISearchIndexSerializer(item) {
    return {
        type: item["type"],
        description: item["description"],
        tags: item["tags"],
        connectionName: item["connectionName"],
        indexName: item["indexName"],
        fieldMapping: !item["fieldMapping"]
            ? item["fieldMapping"]
            : fieldMappingSerializer(item["fieldMapping"]),
    };
}
function azureAISearchIndexDeserializer(item) {
    return {
        type: item["type"],
        id: item["id"],
        name: item["name"],
        version: item["version"],
        description: item["description"],
        tags: item["tags"],
        connectionName: item["connectionName"],
        indexName: item["indexName"],
        fieldMapping: !item["fieldMapping"]
            ? item["fieldMapping"]
            : fieldMappingDeserializer(item["fieldMapping"]),
    };
}
function fieldMappingSerializer(item) {
    return {
        contentFields: item["contentFields"].map((p) => {
            return p;
        }),
        filepathField: item["filepathField"],
        titleField: item["titleField"],
        urlField: item["urlField"],
        vectorFields: !item["vectorFields"]
            ? item["vectorFields"]
            : item["vectorFields"].map((p) => {
                return p;
            }),
        metadataFields: !item["metadataFields"]
            ? item["metadataFields"]
            : item["metadataFields"].map((p) => {
                return p;
            }),
    };
}
function fieldMappingDeserializer(item) {
    return {
        contentFields: item["contentFields"].map((p) => {
            return p;
        }),
        filepathField: item["filepathField"],
        titleField: item["titleField"],
        urlField: item["urlField"],
        vectorFields: !item["vectorFields"]
            ? item["vectorFields"]
            : item["vectorFields"].map((p) => {
                return p;
            }),
        metadataFields: !item["metadataFields"]
            ? item["metadataFields"]
            : item["metadataFields"].map((p) => {
                return p;
            }),
    };
}
function managedAzureAISearchIndexSerializer(item) {
    return {
        type: item["type"],
        description: item["description"],
        tags: item["tags"],
        vectorStoreId: item["vectorStoreId"],
    };
}
function managedAzureAISearchIndexDeserializer(item) {
    return {
        type: item["type"],
        id: item["id"],
        name: item["name"],
        version: item["version"],
        description: item["description"],
        tags: item["tags"],
        vectorStoreId: item["vectorStoreId"],
    };
}
function cosmosDBIndexSerializer(item) {
    return {
        type: item["type"],
        description: item["description"],
        tags: item["tags"],
        connectionName: item["connectionName"],
        databaseName: item["databaseName"],
        containerName: item["containerName"],
        embeddingConfiguration: embeddingConfigurationSerializer(item["embeddingConfiguration"]),
        fieldMapping: fieldMappingSerializer(item["fieldMapping"]),
    };
}
function cosmosDBIndexDeserializer(item) {
    return {
        type: item["type"],
        id: item["id"],
        name: item["name"],
        version: item["version"],
        description: item["description"],
        tags: item["tags"],
        connectionName: item["connectionName"],
        databaseName: item["databaseName"],
        containerName: item["containerName"],
        embeddingConfiguration: embeddingConfigurationDeserializer(item["embeddingConfiguration"]),
        fieldMapping: fieldMappingDeserializer(item["fieldMapping"]),
    };
}
function embeddingConfigurationSerializer(item) {
    return {
        modelDeploymentName: item["modelDeploymentName"],
        embeddingField: item["embeddingField"],
    };
}
function embeddingConfigurationDeserializer(item) {
    return {
        modelDeploymentName: item["modelDeploymentName"],
        embeddingField: item["embeddingField"],
    };
}
function deploymentDeserializer(item) {
    return {
        type: item["type"],
        name: item["name"],
    };
}
function deploymentUnionDeserializer(item) {
    switch (item.type) {
        case "ModelDeployment":
            return modelDeploymentDeserializer(item);
        default:
            return deploymentDeserializer(item);
    }
}
function modelDeploymentDeserializer(item) {
    return {
        type: item["type"],
        name: item["name"],
        modelName: item["modelName"],
        modelVersion: item["modelVersion"],
        modelPublisher: item["modelPublisher"],
        capabilities: item["capabilities"],
        sku: skuDeserializer(item["sku"]),
        connectionName: item["connectionName"],
    };
}
function skuDeserializer(item) {
    return {
        capacity: item["capacity"],
        family: item["family"],
        name: item["name"],
        size: item["size"],
        tier: item["tier"],
    };
}
function _pagedDeploymentDeserializer(item) {
    return {
        value: deploymentUnionArrayDeserializer(item["value"]),
        nextLink: item["nextLink"],
    };
}
function deploymentUnionArrayDeserializer(result) {
    return result.map((item) => {
        return deploymentUnionDeserializer(item);
    });
}
function redTeamSerializer(item) {
    return {
        displayName: item["displayName"],
        numTurns: item["numTurns"],
        attackStrategies: !item["attackStrategies"]
            ? item["attackStrategies"]
            : item["attackStrategies"].map((p) => {
                return p;
            }),
        simulationOnly: item["simulationOnly"],
        riskCategories: !item["riskCategories"]
            ? item["riskCategories"]
            : item["riskCategories"].map((p) => {
                return p;
            }),
        applicationScenario: item["applicationScenario"],
        tags: item["tags"],
        properties: item["properties"],
        target: targetConfigUnionSerializer(item["target"]),
    };
}
function redTeamDeserializer(item) {
    return {
        name: item["id"],
        displayName: item["displayName"],
        numTurns: item["numTurns"],
        attackStrategies: !item["attackStrategies"]
            ? item["attackStrategies"]
            : item["attackStrategies"].map((p) => {
                return p;
            }),
        simulationOnly: item["simulationOnly"],
        riskCategories: !item["riskCategories"]
            ? item["riskCategories"]
            : item["riskCategories"].map((p) => {
                return p;
            }),
        applicationScenario: item["applicationScenario"],
        tags: item["tags"],
        properties: item["properties"],
        status: item["status"],
        target: targetConfigUnionDeserializer(item["target"]),
    };
}
function targetConfigSerializer(item) {
    return { type: item["type"] };
}
function targetConfigDeserializer(item) {
    return {
        type: item["type"],
    };
}
function targetConfigUnionSerializer(item) {
    switch (item.type) {
        case "AzureOpenAIModel":
            return azureOpenAIModelConfigurationSerializer(item);
        default:
            return targetConfigSerializer(item);
    }
}
function targetConfigUnionDeserializer(item) {
    switch (item.type) {
        case "AzureOpenAIModel":
            return azureOpenAIModelConfigurationDeserializer(item);
        default:
            return targetConfigDeserializer(item);
    }
}
function azureOpenAIModelConfigurationSerializer(item) {
    return { modelDeploymentName: item["modelDeploymentName"] };
}
function azureOpenAIModelConfigurationDeserializer(item) {
    return {
        type: item["type"],
        modelDeploymentName: item["modelDeploymentName"],
    };
}
function _pagedRedTeamDeserializer(item) {
    return {
        value: redTeamArrayDeserializer(item["value"]),
        nextLink: item["nextLink"],
    };
}
function redTeamArraySerializer(result) {
    return result.map((item) => {
        return redTeamSerializer(item);
    });
}
function redTeamArrayDeserializer(result) {
    return result.map((item) => {
        return redTeamDeserializer(item);
    });
}
/** Azure AI Projects API versions */
var KnownVersions;
(function (KnownVersions) {
    /** Azure AI API version 2025-05-01. */
    KnownVersions["V20250501"] = "2025-05-01";
    /** Azure AI API version 2025-05-15-preview. */
    KnownVersions["V20250515Preview"] = "2025-05-15-preview";
})(KnownVersions || (exports.KnownVersions = KnownVersions = {}));
