// @azure/ai-projects/dist/react-native/api/connections/operations.js
// Copyright (c) Microsoft Corporation.
// Licensed under the MIT License.
import { __asyncValues } from "tslib";
import { connectionDeserializer, _pagedConnectionDeserializer, } from "../../models/models.js";
import { expandUrlTemplate } from "../../static-helpers/urlTemplate.js";
import { buildPagedAsyncIterator, } from "../../static-helpers/pagingHelpers.js";
import { createRestError, operationOptionsToRequestParameters, } from "@azure-rest/core-client";
export function _listSend(context, options = { requestOptions: {} }) {
    var _a, _b;
    const path = expandUrlTemplate("/connections{?api-version,connectionType,defaultConnection}", {
        "api-version": context.apiVersion,
        connectionType: options === null || options === void 0 ? void 0 : options.connectionType,
        defaultConnection: options === null || options === void 0 ? void 0 : options.defaultConnection,
    }, {
        allowReserved: (_a = options === null || options === void 0 ? void 0 : options.requestOptions) === null || _a === void 0 ? void 0 : _a.skipUrlEncoding,
    });
    return context.path(path).get(Object.assign(Object.assign({}, operationOptionsToRequestParameters(options)), { headers: Object.assign(Object.assign(Object.assign({}, ((options === null || options === void 0 ? void 0 : options.clientRequestId) !== undefined
            ? { "x-ms-client-request-id": options === null || options === void 0 ? void 0 : options.clientRequestId }
            : {})), { accept: "application/json" }), (_b = options.requestOptions) === null || _b === void 0 ? void 0 : _b.headers) }));
}
export async function _listDeserialize(result) {
    const expectedStatuses = ["200"];
    if (!expectedStatuses.includes(result.status)) {
        throw createRestError(result);
    }
    return _pagedConnectionDeserializer(result.body);
}
/** List all connections in the project, without populating connection credentials */
export function list(context, options = { requestOptions: {} }) {
    return buildPagedAsyncIterator(context, () => _listSend(context, options), _listDeserialize, ["200"], { itemName: "value", nextLinkName: "nextLink" });
}
export function _getWithCredentialsSend(context, name, options = { requestOptions: {} }) {
    var _a, _b;
    const path = expandUrlTemplate("/connections/{name}/getConnectionWithCredentials{?api-version}", {
        name: name,
        "api-version": context.apiVersion,
    }, {
        allowReserved: (_a = options === null || options === void 0 ? void 0 : options.requestOptions) === null || _a === void 0 ? void 0 : _a.skipUrlEncoding,
    });
    return context.path(path).post(Object.assign(Object.assign({}, operationOptionsToRequestParameters(options)), { headers: Object.assign(Object.assign(Object.assign({}, ((options === null || options === void 0 ? void 0 : options.clientRequestId) !== undefined
            ? { "x-ms-client-request-id": options === null || options === void 0 ? void 0 : options.clientRequestId }
            : {})), { accept: "application/json" }), (_b = options.requestOptions) === null || _b === void 0 ? void 0 : _b.headers) }));
}
export async function _getWithCredentialsDeserialize(result) {
    const expectedStatuses = ["200"];
    if (!expectedStatuses.includes(result.status)) {
        throw createRestError(result);
    }
    return connectionDeserializer(result.body);
}
/** Get a connection by name, with its connection credentials */
export async function getWithCredentials(context, name, options = { requestOptions: {} }) {
    const result = await _getWithCredentialsSend(context, name, options);
    return _getWithCredentialsDeserialize(result);
}
export function _getSend(context, name, options = { requestOptions: {} }) {
    var _a, _b;
    const path = expandUrlTemplate("/connections/{name}{?api-version}", {
        name: name,
        "api-version": context.apiVersion,
    }, {
        allowReserved: (_a = options === null || options === void 0 ? void 0 : options.requestOptions) === null || _a === void 0 ? void 0 : _a.skipUrlEncoding,
    });
    return context.path(path).get(Object.assign(Object.assign({}, operationOptionsToRequestParameters(options)), { headers: Object.assign(Object.assign(Object.assign({}, ((options === null || options === void 0 ? void 0 : options.clientRequestId) !== undefined
            ? { "x-ms-client-request-id": options === null || options === void 0 ? void 0 : options.clientRequestId }
            : {})), { accept: "application/json" }), (_b = options.requestOptions) === null || _b === void 0 ? void 0 : _b.headers) }));
}
export async function _getDeserialize(result) {
    const expectedStatuses = ["200"];
    if (!expectedStatuses.includes(result.status)) {
        throw createRestError(result);
    }
    return connectionDeserializer(result.body);
}
/** Get a connection by name, without populating connection credentials */
export async function get(context, name, includeCredentials, options = { requestOptions: {} }) {
    if (includeCredentials) {
        return getWithCredentials(context, name, options);
    }
    const result = await _getSend(context, name, options);
    return _getDeserialize(result);
}
/**
 * Get the default connection for a given connection type.
 *
 * @param context - The AIProjectContext client
 * @param connectionType - The type of the connection. Required.
 * @param includeCredentials - Whether to include credentials in the response. Default is false.
 * @returns A Connection object
 * @throws Error if no default connection is found for the given type.
 */
export async function getDefault(context, connectionType, includeCredentials = false) {
    var _a, e_1, _b, _c;
    const listOptions = {
        connectionType,
        defaultConnection: true,
    };
    // Use the list function to find default connections of the specified type
    const connections = list(context, listOptions);
    try {
        // Find the first default connection
        for (var _d = true, connections_1 = __asyncValues(connections), connections_1_1; connections_1_1 = await connections_1.next(), _a = connections_1_1.done, !_a; _d = true) {
            _c = connections_1_1.value;
            _d = false;
            const connection = _c;
            if (includeCredentials) {
                // If credentials are requested, get the connection with credentials
                return getWithCredentials(context, connection.name);
            }
            return connection;
        }
    }
    catch (e_1_1) { e_1 = { error: e_1_1 }; }
    finally {
        try {
            if (!_d && !_a && (_b = connections_1.return)) await _b.call(connections_1);
        }
        finally { if (e_1) throw e_1.error; }
    }
    throw new Error(`No default connection found for type: ${connectionType}.`);
}
