// @azure/ai-agents/dist/esm/classic/runs/index.js
// Copyright (c) Microsoft Corporation.
// Licensed under the MIT License.
import { cancelRun, submitToolOutputsToRun, updateRun, getRun, listRuns, createRun, createRunAndPoll, } from "../../api/runs/operations.js";
import { createThreadAndRun } from "../../api/operations.js";
function _getRuns(context) {
    return {
        cancel: (threadId, runId, options) => cancelRun(context, threadId, runId, options),
        submitToolOutputs: (threadId, runId, toolOutputs, options) => submitToolOutputsToRun(context, threadId, runId, toolOutputs, options),
        update: (threadId, runId, options) => updateRun(context, threadId, runId, options),
        get: (threadId, runId, options) => getRun(context, threadId, runId, options),
        list: (threadId, options) => listRuns(context, threadId, options),
        create: (threadId, assistantId, options) => createRun(context, threadId, assistantId, options),
        createAndPoll: (threadId, assistantId, options) => createRunAndPoll(context, threadId, assistantId, options),
        createThreadAndRun: (assistantId, options) => createThreadAndRun(context, assistantId, options),
    };
}
export function _getRunsOperations(context) {
    return Object.assign({}, _getRuns(context));
}
