// @azure/ai-agents/dist/commonjs/api/runSteps/operations.js
"use strict";
// Copyright (c) Microsoft Corporation.
// Licensed under the MIT License.
Object.defineProperty(exports, "__esModule", { value: true });
exports._listRunStepsSend = _listRunStepsSend;
exports._listRunStepsDeserialize = _listRunStepsDeserialize;
exports.listRunSteps = listRunSteps;
exports._getRunStepSend = _getRunStepSend;
exports._getRunStepDeserialize = _getRunStepDeserialize;
exports.getRunStep = getRunStep;
const models_js_1 = require("../../models/models.js");
const pagingHelpers_js_1 = require("../../static-helpers/pagingHelpers.js");
const urlTemplate_js_1 = require("../../static-helpers/urlTemplate.js");
const core_client_1 = require("@azure-rest/core-client");
function _listRunStepsSend(context, threadId, runId, options = { requestOptions: {} }) {
    var _a, _b;
    const path = (0, urlTemplate_js_1.expandUrlTemplate)("/threads/{threadId}/runs/{runId}/steps{?include%5B%5D,api%2Dversion,limit,order,after,before}", {
        threadId: threadId,
        runId: runId,
        "include%5B%5D": !(options === null || options === void 0 ? void 0 : options.include)
            ? options === null || options === void 0 ? void 0 : options.include
            : options === null || options === void 0 ? void 0 : options.include.map((p) => {
                return p;
            }),
        "api%2Dversion": context.apiVersion,
        limit: options === null || options === void 0 ? void 0 : options.limit,
        order: options === null || options === void 0 ? void 0 : options.order,
        after: options === null || options === void 0 ? void 0 : options.after,
        before: options === null || options === void 0 ? void 0 : options.before,
    }, {
        allowReserved: (_a = options === null || options === void 0 ? void 0 : options.requestOptions) === null || _a === void 0 ? void 0 : _a.skipUrlEncoding,
    });
    return context.path(path).get(Object.assign(Object.assign({}, (0, core_client_1.operationOptionsToRequestParameters)(options)), { headers: Object.assign({ accept: "application/json" }, (_b = options.requestOptions) === null || _b === void 0 ? void 0 : _b.headers) }));
}
async function _listRunStepsDeserialize(result) {
    const expectedStatuses = ["200"];
    if (!expectedStatuses.includes(result.status)) {
        throw (0, core_client_1.createRestError)(result);
    }
    return (0, models_js_1._agentsPagedResultRunStepDeserializer)(result.body);
}
/** Gets a list of run steps from a thread run. */
function listRunSteps(context, threadId, runId, options = { requestOptions: {} }) {
    return (0, pagingHelpers_js_1.buildPagedAsyncIterator)(context, () => _listRunStepsSend(context, threadId, runId, options), _listRunStepsDeserialize, ["200"], { itemName: "data" });
}
function _getRunStepSend(context, threadId, runId, stepId, options = { requestOptions: {} }) {
    var _a, _b;
    const path = (0, urlTemplate_js_1.expandUrlTemplate)("/threads/{threadId}/runs/{runId}/steps/{stepId}{?api%2Dversion,include%5B%5D}", {
        threadId: threadId,
        runId: runId,
        stepId: stepId,
        "api%2Dversion": context.apiVersion,
        "include%5B%5D": !(options === null || options === void 0 ? void 0 : options.include)
            ? options === null || options === void 0 ? void 0 : options.include
            : options === null || options === void 0 ? void 0 : options.include.map((p) => {
                return p;
            }),
    }, {
        allowReserved: (_a = options === null || options === void 0 ? void 0 : options.requestOptions) === null || _a === void 0 ? void 0 : _a.skipUrlEncoding,
    });
    return context.path(path).get(Object.assign(Object.assign({}, (0, core_client_1.operationOptionsToRequestParameters)(options)), { headers: Object.assign({ accept: "application/json" }, (_b = options.requestOptions) === null || _b === void 0 ? void 0 : _b.headers) }));
}
async function _getRunStepDeserialize(result) {
    const expectedStatuses = ["200"];
    if (!expectedStatuses.includes(result.status)) {
        throw (0, core_client_1.createRestError)(result);
    }
    return (0, models_js_1.runStepDeserializer)(result.body);
}
/** Retrieves a single run step from a thread run. */
async function getRunStep(context, threadId, runId, stepId, options = { requestOptions: {} }) {
    const result = await _getRunStepSend(context, threadId, runId, stepId, options);
    return _getRunStepDeserialize(result);
}
