// @azure/ai-projects/dist/commonjs/api/datasets/index.js
"use strict";
// Copyright (c) Microsoft Corporation.
// Licensed under the MIT License.
Object.defineProperty(exports, "__esModule", { value: true });
exports.listVersions = exports.list = exports.get = exports.$delete = exports.createOrUpdate = exports.pendingUpload = exports.getCredentials = void 0;
var operations_js_1 = require("./operations.js");
Object.defineProperty(exports, "getCredentials", { enumerable: true, get: function () { return operations_js_1.getCredentials; } });
Object.defineProperty(exports, "pendingUpload", { enumerable: true, get: function () { return operations_js_1.pendingUpload; } });
Object.defineProperty(exports, "createOrUpdate", { enumerable: true, get: function () { return operations_js_1.createOrUpdate; } });
Object.defineProperty(exports, "$delete", { enumerable: true, get: function () { return operations_js_1.$delete; } });
Object.defineProperty(exports, "get", { enumerable: true, get: function () { return operations_js_1.get; } });
Object.defineProperty(exports, "list", { enumerable: true, get: function () { return operations_js_1.list; } });
Object.defineProperty(exports, "listVersions", { enumerable: true, get: function () { return operations_js_1.listVersions; } });
